### PR TITLE
run notebooks on coast repo

### PR DESCRIPTION
**Pull request by thogar-computer**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been reviewed and added/updated if needed (for bug fixes/features)
    - Some docs updates need to be made in the `COAsT-site` repo, in a separate PR. See [contributing to documentation ](https://british-oceanographic-data-centre.github.io/COAsT/docs/contributing-docs/) for details.

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [X] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

The PO would like the notebooks to be run within the GitHub workflow

Issue URL: #552 

## What is the new behavior?

All the notebook GitHub actions have been copied over and slimed down to only run the notebooks (not moving the markdown and Images) 

### Checklist
- [x] Code compiles correctly
- [x] Created tests which fail without the change (if possible)
- [x] All tests passing (`pip install . && pytest unit_testing/unit_test.py -s`)
- [x] Extended the README/documentation, if necessary
- [x] Added myself / the copyright holder to the AUTHORS file

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

**Comment by thogar-computer** [comment]
We spilt out the notebook runner/converts in the coast-site repo so I have keep it the same here.... not sure if that'll mean a quicker run / better logging or if it just means a lot of repeated (wasted) compute...

File: .github/workflows/markdown-altimetry.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+          conda info
+          conda list
+          conda config --show
+          cd $GITHUB_WORKSPACE/external
+          sudo apt install wget unzip -y
+          wget -q https://linkedsystems.uk/erddap/files/COAsT_example_files/COAsT_example_files.zip
+          unzip COAsT_example_files.zip && mv COAsT_example_files ./example_scripts/notebook_tutorials/runnable_notebooks/altimetry/example_files
+          ls /home/runner/work/COAsT-site/COAsT-site/external/example_scripts/notebook_tutorials/runnable_notebooks/altimetry/example_files
+          mkdir -p ./example_scripts/notebook_tutorials/markdown/altimetry
+          ls ./example_scripts/notebooks
```
Comment (roje-bodc) [inline .github/workflows/markdown-altimetry.yml:47]: Just  a reminder to update this path to `./example_scripts/notebook_tutorials` for each of the workflows :)

File: .github/workflows/markdown-altimetry.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+name: notebook_to_md (altimetry)
+
+on:
+  # External webhook trigger
+  repository_dispatch:
+   types: [push-notebook]
```
Comment (roje-bodc) [inline .github/workflows/markdown-altimetry.yml:6]: We probably want to change the trigger for this. This webhook approach was used to trigger the notebook action on the coast-site repo once a merge to develop was done on the coast repo,

Comment (thogar-computer) [reply .github/workflows/markdown-altimetry.yml]: @roje-bodc  changed to on pull request

**Pull request merged**